tests/test_header_matching.js

```diff
@@ -95,6 +95,21 @@ test('match headers with function', async t => {
   scope.done()
 })
 
+test('match headers with function and allowUnmocked', async t => {
```

Comment:
If you merge master into this file, there are some new tests which are a bit clearer and more specific. If you'd like, you could update this to use one of those.